src/celeritas/neutron/interactor/NeutronElasticInteractor.hh

```diff
@@ -0,0 +1,156 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/interactor/NeutronElasticInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/ArrayOperators.hh"
+#include "celeritas/Constants.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/neutron/data/NeutronElasticData.hh"
+#include "celeritas/phys/FourVector.hh"
+#include "celeritas/phys/Interaction.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+#include "detail/InvariantQ2Sampler.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform neutron elastic scattering based on the CHIPS (Chiral invariant
+ * phase space) model.
+ *
+ * \note This performs the sampling procedure as in G4HadronElastic,
+ * G4ChipsElasticModel and G4ChipsNeutronElasticXS, as partly documented
+ * in section 21.1.3 of the Geant4 Physics Reference (release 11.2).
+ */
+class NeutronElasticInteractor
```

Comment:
Should this be `ChipsInteractor` or `ChipsNeutronInteractor` or something? Will there be other neutron elastic scattering models? (In our neutron transport code, for elastic scattering above the thermal energy range and below 20 MeV, we use a super-simple point kinematic elastic scattering model...)

Reply:
Actually, a good point for future extensions! The better choice may be `ChipsNeutronElasticModel` and `ChipsNeutronElasticInteractor` (as CHIPS supports not only neutron elastic, but other particles and inelastic interactions (In Geant4, Chips inelastic is still in use for Kaons at the high energy region).   One downside is that class names are getting longer for the combination of a specific particle and a process.  I will rename them as above (later today) since we may extend the elastic scattering for proton too.  Of course, we can always rename and restructure codes as more physics are added.  

The `neutron-HP` extension below 20MeV may be a challenging project for us as there are interference between the scattering  process and radiative decays (i.e., de/excitation) with different atomic levels, especially for heavy nuclei (which could have thousands or millions of combination for possible transitions) - may be a good project after retirement (even for you?).  Yes, we can enjoy the elastic scattering with billiard balls whenever physics need is classical.

Reply:
So the interactor implementation for incident neutrons in the CHIPS model is substantially different from other hadrons?

Reply:
Actually not for the elastic scattering (the code structure is quite similar for different particles with different sets of XS data), but may be for inelastic (which I have not looked much as most of CHIPS inelastic channels are now obsolete in Geant4 and we are going to implement the Bertini cascade instead).  As you know, the proton scattering is not considered at this point (even though it may be easy to add once this MR is done) as it requires other processes like eloss, msc and etc.